lib/stream.js

```diff
@@ -23,17 +22,14 @@ export const makeAllStream = (spawned, {all}) => {
 };
 
 // On failure, `result.stdout|stderr|all` should contain the currently buffered stream
+// They are automatically closed and flushed by Node.js when the child process exits
+// We guarantee this by calling `childProcess.kill()`
 const getBufferedData = async (stream, streamPromise) => {
 	// When `buffer` is `false`, `streamPromise` is `undefined` and there is no buffered data to retrieve
 	if (!stream || streamPromise === undefined) {
 		return;
 	}
 
-	// Wait for the `all` stream to receive the last chunk before destroying the stream
-	await setTimeout(0);
-
-	stream.destroy();
```

Comment:
`childProcess.spawn()` automatically destroys `childProcess.stdin|stdout|stderr` on exit. Destroying it manually prevents proper flushing.

`await setTimeout(0)` was a hack to "fix" that improper flushing, which manifested itself more with `childProcess.all` since that stream involves an additional stream, therefore an additional delay.

The destroying logic originates from 8 years ago, in the following PR: https://github.com/sindresorhus/execa/pull/68, which itself originates from the following issue: https://github.com/sindresorhus/execa/issues/32

> Right now we aren't destroying those streams. Probably should.

We now have 600+ tests, which work properly without that logic. From checking Node.js current code, that cleanup logic is not necessary, and prevents proper flushing.